File: eecs268/2020summer/video-lectures/2020.06.11.debugger/Queue.cpp

template <typename T>
Queue<T>::Queue()
{
	m_front = nullptr;
	m_back = nullptr;
}

template <typename T>
void Queue<T>::enqueue(T entry)
{
	//what if we're adding to an empty Queue?
	if(isEmpty())
	{
		//this entry is at the front and
		//the back of the queue. Deal
		//with both pointers
		m_back = new Node<T>(entry);
		m_front = m_back;
	}
	else
	{
		//just update what's at the back
		Node<T>* temp = m_back;
		m_back = new Node<T>(entry);
		temp->setNext(m_back);
	}
}

template <typename T>
bool Queue<T>::isEmpty() const
{
	return(m_front==nullptr);
}

File: eecs268/2022spring/MWF/2022.02.14.queues/queue.py
from node import Node

class Queue:
    def __init__(self):
        self._front = None
        self._back =  None

    def is_empty(self):
        return self._front == None

    def enqueue(self, entry):
        new_back = Node(entry)
        if self.is_empty():
            self._front = new_back
            self._back = new_back
        else:
            self._back.next = new_back
            self._back = new_back
            self._back.next = None

    def dequeue(self):
        #board work: define dequeue
        #return the value (if possible)
        

    
